mempool/src/mempool_service.rs

```diff
@@ -27,28 +29,21 @@ pub(crate) struct MempoolService {
     pub(crate) core_mempool: Arc<Mutex<CoreMempool>>,
 }
 
+#[tonic::async_trait]
 impl Mempool for MempoolService {
-    fn add_transaction_with_validation(
-        &mut self,
-        ctx: ::grpcio::RpcContext<'_>,
-        req: crate::proto::mempool::AddTransactionWithValidationRequest,
-        sink: ::grpcio::UnarySink<crate::proto::mempool::AddTransactionWithValidationResponse>,
-    ) {
+    async fn add_transaction_with_validation(
+        &self,
+        request: tonic::Request<AddTransactionWithValidationRequest>,
+    ) -> Result<tonic::Response<AddTransactionWithValidationResponse>, tonic::Status> {
+        let req = request.into_inner();
         trace!("[GRPC] Mempool::add_transaction_with_validation");
```

Comment:
You probably want to change the message here since you are replacing GRPC.

Reply:
I'm not replacing grpc, its still grpc its just using a different library.